Updating TinyGo support (#174)
fermyon/developer: content/spin/go-components.md

## Versions

> TinyGo requires Go versions `1.15.x` or newer. The recommendation is to use Go version `1.19.x`, and TinyGo version `0.25.x`. TinyGo version `0.26.x` is currently not supported with Spin.

Reviewer: small correction tinygo 0.24+ requires go 1.16+
https://github.com/tinygo-org/tinygo/blob/release/CHANGELOG.md#0240

Reviewer 2: ```suggestion
TinyGo `0.25.x` is recommended, which requires Go `1.16.x` or newer.

> TinyGo version `0.26.x` is known to have issues with Spin on some systems.
```
